src/api/libcellml/analyserequationast.h

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include "libcellml/analyser.h"
+#include "libcellml/generator.h"
+
+#undef NAN
+
+namespace libcellml {
+
+/**
+ * @brief The AnalyserEquationAst class.
+ *
+ * The AnalyserEquationAst class is for representing an equation AST in the
```

Comment:
Do we actually say that AST means abstract syntax tree anywhere?

Reply:
No, we don't, hence I have fixed this in commit 1983a4c.